docs/src/airdrop.md

```diff
@@ -1,29 +1,87 @@
-  
-* Prevents claimee from claiming more than eligible for. Claimees receive the requested claim amount minus any amount claimed so far.
+- Merkle Tree Validation System: Uses proofs to verify claims ensuring that each participant can only claim their tokens. The contract determines the submitted Merkle proof against all previously added roots. If a match is found, indicating eligibility, the contract proceeds with the distribution.
+- Event Emission: The contract emits a `Claimed` event for each successful claim. This event includes the claimee's address and the amount received, enabling easy monitoring of claims.
+- ```
+  self.emit(Claimed { address: claimee, received: to_mint });
+
+  ```
 
 ## Setup
 
-* Protocols intending to use the Konoha airdrop contract to distribute airdrops should deploy its instance of the contract.
-  
-* Set the merkle root: When a new round of airdrops is available, add the merkle root to the contract to validate claims.
-  
-* Claimees can claim their tokens by providing their address, claimable amount and proof.
-  
+- Protocols intending to use the Konoha airdrop component to distribute airdrops should integrate its instance within their deployment process.
+- Set the merkle root: When a new round of airdrops is available, add the merkle root to the contract to validate claims.
+- Claimees can claim their tokens by providing their address, claimable amount and proof.
+
+## Deployment
+
+To deploy the Konoha airdrop component, you should follow these steps:
+
+- You should ensure that your Starknet environment is set up correctly and you have the necessary tools installed.
+- The `airdrop` module needs to be part of your Konoha deployment setup.You should ensure it is included in your building process.
+
 ## Usage
 
-The Konoha airdrop component is designed to work within the framework and it offers various customization options to fit each protocol needs.
+The Konoha airdrop component is designed to integrate into your protocol without any dependencies on other Konoha components. This means it can be used standalone within the Konoha ecosystem.
+
+- Merkle tree: You can take control of the tree by modifying the code to fit your protocol's needs.
+- Event handling: You can add or modify events to better monitor the distribution of tokens to claimees and the amount of token being distributed.
+- Governance token dispatcher: You can implement your token dispatch logic if your protocol uses a custom token standard.
+  While the airdrop component is designed to work within the Konoha ecosystem, you can adapt it for use outside Konoha. However, you will need to ensure the contract interfaces match those of your deployment environment.
+
+## Generating the Merkle Tree Root
+
+To generate the Merkle tree root, you need to create a tree structure where each leaf is a hash of an airdrop claim and compute the root of the tree.
```

Comment:
I would link the carmine-api and defispring backend directly. And explain whether they are compatible, what changes need to be done to the defispring backend if any.